tests/ui/test_ui_layer.py

```diff
@@ -0,0 +1,56 @@
+from unittest import mock
+
+import numpy as np
+import pytest
+from mpi4py import MPI
+
+from httomo.ui_layer import UiLayer
+
+
+def test_pipeline1_py(standard_data, python_pipeline1):
+    """Testing existing python pipelines by reading them and generating Pipelines
+    """
+
+    comm = MPI.COMM_WORLD
+    LayerUI = UiLayer(python_pipeline1, standard_data, comm=comm)
+
+    pipeline = LayerUI.build_pipeline()
+
+    assert LayerUI.PipelineStageConfig[0]['method'] == 'standard_tomo'
+    assert LayerUI.PipelineStageConfig[0]['module_path'] == 'httomo.data.hdf.loaders'
+    assert LayerUI.PipelineStageConfig[1]['method'] == 'normalize'
+    assert LayerUI.PipelineStageConfig[1]['module_path'] == 'tomopy.prep.normalize'
+    assert LayerUI.PipelineStageConfig[2]['method'] == 'minus_log'
+    assert LayerUI.PipelineStageConfig[2]['module_path'] == 'tomopy.prep.normalize'
+    assert LayerUI.PipelineStageConfig[3]['method'] == 'find_center_vo'
+    assert LayerUI.PipelineStageConfig[3]['module_path'] == 'tomopy.recon.rotation'
+    assert LayerUI.PipelineStageConfig[3]['side_outputs'] == {'cor' : 'centre_of_rotation'}
+    assert LayerUI.PipelineStageConfig[4]['method'] == 'recon'
+    assert LayerUI.PipelineStageConfig[4]['module_path'] == 'tomopy.recon.algorithm'
+    assert LayerUI.PipelineStageConfig[5]['method'] == 'save_to_images'
+    assert LayerUI.PipelineStageConfig[5]['module_path'] == 'httomolib.misc.images'
+    assert len(pipeline._methods) == 5
```

Comment:
Would it be better to have the length assertion before the other assertions? If the length of the list is shorter than expected, one of the other assertions is going to throw an `IndexError`. In which case the test will fail, but it'll be because of an error in the test itself rather than because of an assertion error?

It's not really a big deal because the test will fail as a consequence of the list being an incorrect length (which is part of what the test is aiming for I guess), but the test output may not look as clear about what caused the failure if the test itself has an `IndexError` occurring.